### ISSUE TITLE
Display details of nested errors

### ISSUE THREAD

**Issue by sineed** [jonasschmidt/ex_json_schema#16]
I ran several validations against my schemata and found some problems related to for errors of `anyOf` keyword. For example I have the following rule inside my schema:
```json
"included": {
  "type": "array",
    "items": {
      "anyOf": [
        { "$ref": "data/image-data.json" },
        { "$ref": "data/partner-data.json" }
      ]
    }
  }
```

refs can contain a lot of additional validation rules. But when I perform json validation against invalid data it says only:

```elixir
{"Expected any of the schemata to match but none did.", "#/included/0"}
```
It may be better to list errors against each of validation object (in my case 2 lists).

I don't use `oneOf` or `allOf` keywords but suppose that they have the same behaviour.

I can make a PR that adds more info about these errors

**Comment by jonasschmidt** [comment]
Yes, you're absolutely right. I already had this one on my internal todo list but never got to it. It shouldn't be hard to change, so I'd be happy if you could provide a PR.

**Comment by jonasschmidt** [comment]
Improved error presentation and formatting is released in `0.6.0`. The README has some examples of how to use it. I will close this issue for now.
